plugins/modules/discovery_rule.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2017, Matthias M Dellweg <dellweg@atix.de> (ATIX AG)
+# (c) 2022, Jeffrey van Pelt <jeff@vanpelt.one>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: discovery_rule
+version_added: x.x.x
+short_description: Manage Host Discovery Rules
+description:
+  - Manage Host Discovery Rules
+author:
+  - "Jeffrey van Pelt (@Thulium-Drake)"
+options:
+  name:
+    description:
+      - Name of the Discovery Rule
+    required: True
+    type: str
+  search:
+    description:
+      - Expression to match newly discovered hosts with
+    required: True
+    type: str
+  hostgroup:
+    description:
+      - Hostgroup to assign hosts to
+    required: True
+    type: str
+  hostname:
+    description:
+      - Hostname to assign to discovered host(s)
+      - When matching multiple hosts, must provide unique hostnames for each of the discovered hosts
+    type: str
+  enabled:
+    description:
+      - Enable or disable the rule
+    type: bool
+  priority:
+    description:
+      - Priority of the rule
+    type: int
+  max_count:
+    description:
+      - Maximum amount of hosts to provision with the rule
+      - 0 means no limit
+    type: int
+extends_documentation_fragment:
+  locations:
+    description:
+      - List of locations the discovery rule should be assigned to
+    type: list
+    elements: str
+  organizations:
+    description:
+      - List of organizations the discovery rule should be assigned to
+    type: list
+    elements: str
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state
+  - theforeman.foreman.foreman.nested_parameters
```

Comment:
```suggestion
  - theforeman.foreman.foreman.taxonomy
```

We don't use nested params anymore, but we do have taxonomy! :)

Reply:
Check, done!